contribs/src/main/java/com/netflix/conductor/contribs/kafka/KafkaLoginModule.java

```diff
@@ -0,0 +1,131 @@
+package com.netflix.conductor.contribs.kafka;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Java security module used to provide JAAS username and password for access to a Kafka topic
+ * @author rickfish
+ *
+ */
+public class KafkaLoginModule implements LoginModule {
```

Comment:
@rickfish Can you elaborate how this class will be leveraged to login into Kafka? Conductor runs on Jetty by default . Is this class expected to be called by the Java application server?

Reply:
This is used by JAAS. In the init() method of  KafkaObservableQueue it will load  a JAAS config file if it is  specified as a property. If specified, you reference the login module to use to get the userid/pw. See contribs/src/main/resources for examples of  these JAAS config files. I think security was left out of the other Kafka stuff because it is so convoluted.